src/tee/codemirrorPlugins/dragAndDropImages.ts

```diff
@@ -0,0 +1,74 @@
+import { EditorView, ViewPlugin } from "@codemirror/view";
+import { EditorSelection } from "@codemirror/state";
+
+type DragAndDropImagePluginConfig = {
+  // createImageReference is called whenever an image is dropped into the editor
+  // the handle can return a string that will be inserted at the the drop position
+  createImageReference: (file: File) => Promise<string | undefined>;
+};
+
+export const dragAndDropImagesPlugin = ({
+  createImageReference,
+}: DragAndDropImagePluginConfig) => {
+  let view: EditorView;
+
+  let previousSelection: EditorSelection;
+
+  const onDragEnter = (event) => {
+    previousSelection = view.state.selection;
+
+    event.preventDefault(); // cancel event to indicate drag is allowed
+    event.dataTransfer.dropEffect = "copy"; // Show a visual cue that a copy operation is happening
+  };
+
+  const onDragOver = (event) => {
+    event.preventDefault(); // cancel event to indicate drag is allowed
+  };
+
+  const onDrop = (event) => {
+    const files = event.dataTransfer.files;
+    if (files.length > 0) {
+      const file = files[0];
+      if (file.type.startsWith("image/")) {
```

Comment:
is this a reliable check for images that can be displayed on the web? I feel like we should limit the formats more specifically to a few known good ones.